compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -59,7 +60,10 @@ class ProtobufGenerator(
       }
       .add(s"object $name extends ${e.companionExtends.mkString(" with ")} {")
       .indent
-      .add(s"sealed trait ${e.recognizedEnum.nameSymbol} extends $name")
+      .add(
+        s"sealed abstract class ${e.recognizedEnum.nameSymbol}(override val value: _root_.scala.Int) extends $name(value)"
```

Comment:
Is the change here necessary to implement `asRecognized`? (prefer not to break compatibility if not absolutely have to)

Reply:
you're right, I've amended the PR
maybe we can come back to this before the next breaking change release of ScalaPB